interceptors.go

```diff
@@ -0,0 +1,43 @@
+package sarama
+
+// ProducerInterceptor allows you to intercept (and possibly mutate) the records
+// received by the producer before they are published to the Kafka cluster.
+// https://cwiki.apache.org/confluence/display/KAFKA/KIP-42%3A+Add+Producer+and+Consumer+Interceptors#KIP42:AddProducerandConsumerInterceptors-Motivation
+type ProducerInterceptor interface {
```

Comment:
do we need a `ProducerInterceptor` and `ConsumerInterceptor`? could we just have an `Interceptor` with an `onCall(message)`?

Reply:
we would need at least 2 methods, one that receives a `ProducerMessage` and other for `ConsumerMessage`, and that might be an issue, as if someone would like to only intercept a producer, it'll need to implement the `onSend` for consumer too.

So I think it's safe to leave it in 2 separated interfaces